### PR TITLE
Bug 1852305 - Add review grade info to all error states

### PR DESCRIPTION
**Pull request by MozillaNoah**
TLDR: The changes in this PR
- Made a `ProductAnalysisError` Composable to serve as the container for the error state UI to mirror `ProductAnalysis`.
- Refactored the review grade info UI to its own file.
- Added the review grade info to `ProductAnalysisError`

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1852305

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityInfoCard.kt [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            color = FirefoxTheme.colors.textPrimary,
+            style = FirefoxTheme.typography.body2,
+        )
+
+        val link = stringResource(R.string.review_quality_check_info_learn_more_link)
+        val text = stringResource(R.string.review_quality_check_info_learn_more, link)
+        val linkStartIndex = text.indexOf(link)
+        val context = LocalContext.current
+        val linkEndIndex = linkStartIndex + link.length
+        ClickableSubstringLink(
```
Comment (HarrisonOg) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityInfoCard.kt:77]: We're trying to move away from `ClickableSubstringLink` and move to `LinkText` for future clickable links in text blocks.

Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityInfoCard.kt:77]: Yeah, I think since it was officially `@Deprecated` it, there is a ticket to remove all of the usages, but not 100% that was made.

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    modifier: Modifier = Modifier,
+    onReviewGradeLearnMoreClick: (String) -> Unit,
+) {
+    Column(modifier = modifier) {
+        // Error UI to be done in Bug 1840113
+
+        ReviewQualityInfoCard(
+            onLearnMoreClick = onReviewGradeLearnMoreClick,
+        )
+    }
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt:35]: Should we add the Settings card and "powered by.." caption  as well as that's also shown in error states

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+@Composable
+@LightDarkPreview
+private fun ReviewQualityInfoPreview() {
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt:40]: `ProductAnalysisErrorPreview`

**Comment by rahulsainani** [comment]
Looks good overall, a couple of comments for you to consider.

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt (hunk 2 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt:35]: I was originally gonna defer that to the official bug, but I'm cool with adding those now 👍

Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt:35]: Is the powered by caption implemented yet? I'm not seeing it anywhere. Is there a ticket for that?

Comment (calumozilla) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt:35]: Not yet, I'm guessing there's no tickets created for the various UI changes from Verdi's video

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckSettingsCard.kt [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            productRecommendationsEnabled = productRecommendationsEnabled,
+            onProductRecommendationsEnabledStateChange = onProductRecommendationsEnabledStateChange,
+            onTurnOffReviewQualityCheckClick = onTurnOffReviewQualityCheckClick,
+            modifier = Modifier.fillMaxWidth(),
+        )
+    }
+}
+
+@Composable
+private fun SettingsContent(
```
Comment (rahulsainani) [inline fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckSettingsCard.kt:55]: Not sure we need an extra function indirection with the same shape/params, curious what was the thought process to add this? Also, removing the need for an extra preview.

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt (hunk 2 above)
Comment (rahulsainani) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt:53]: we had it under highlights card but it seems it's been removed 🤷

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckSettingsCard.kt (hunk 4 above)
Comment (rahulsainani) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckSettingsCard.kt:55]: Ahh gotcha, makes sense! thanks for the explanation!

**Review by rahulsainani — APPROVED** [review]
Looks good! ✨  Thanks for moving the settings card as well.

File: fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt (hunk 2 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysisError.kt:53]: https://bugzilla.mozilla.org/show_bug.cgi?id=1852994
